repository-commons/src/main/java/io/aiven/elasticsearch/repositories/RepositoryStorageIOProvider.java

```diff
@@ -63,15 +65,16 @@ public RepositoryStorageIOProvider(final String repositoryType,
     public StorageIO createStorageIO(final String basePath, final Settings repositorySettings) throws IOException {
         checkSettings(repositoryType, BUCKET_NAME, repositorySettings);
         final var bucketName = BUCKET_NAME.get(repositorySettings);
+        final var bufferSize = Math.toIntExact(BUFFER_SIZE_SETTING.get(repositorySettings).getBytes());
         Permissions.doPrivileged(() -> createOrRestoreEncryptionKey(basePath, bucketName));
-        return createStorageIOFor(bucketName, new CryptoIOProvider(encryptionKey));
+        return createStorageIOFor(bucketName, new CryptoIOProvider(encryptionKey, bufferSize));
     }
 
     private void createOrRestoreEncryptionKey(final String basePath, final String bucketName) throws IOException {
         if (Objects.isNull(encryptionKey)) {
             final var repositoryMetadataFilePath = basePath + REPOSITORY_METADATA_FILE_NAME;
             final var encKeyRepoMetadata =
-                    createStorageIOFor(bucketName, new CryptoIOProvider(null) {
+                    createStorageIOFor(bucketName, new CryptoIOProvider(null, 0) {
```

Comment:
Hm.. I have some doubts it should be 0 here :) Or I'm missing something?

Reply:
Yes you are right. But in this case `RepositoryStorageIOProvider` restores/creates an encryption key. Ans since we do not need real `CryptoIOProvider` here it is just a stub with overridden 2 methods: `compressAndEncrypt` and `decryptAndDecompress`, which just read/write directly to the storage.

Reply:
Got it. Would you please leave a comment about this?

Reply:
true ... makes sense